src/app/messages/MessageProvider.tsx

```diff
@@ -0,0 +1,61 @@
+import { shouldPolyfill } from '@formatjs/intl-pluralrules/should-polyfill';
+import React, { useEffect, useState } from 'react';
+import { IntlShape, RawIntlProvider } from 'react-intl';
+import { useSelector } from 'react-redux';
+import { book as bookSelector } from '../content/selectors';
+import { match as matchSelector } from '../navigation/selectors';
+import { assertDocument } from '../utils/browser-assertions';
+import createIntl from './createIntl';
+
+// https://formatjs.io/docs/polyfills/intl-pluralrules/#dynamic-import--capability-detection
+async function polyfill(locale: string) {
+  if (shouldPolyfill()) {
+    await import('@formatjs/intl-pluralrules/polyfill');
+  }
+
+  // boolean added by the polyfill
+  if ((Intl.PluralRules as (typeof Intl.PluralRules & {polyfilled?: boolean})).polyfilled) {
+    await import(`@formatjs/intl-pluralrules/locale-data/${locale}`);
+  }
+}
+
+// tslint:disable-next-line:variable-name
+const MessageProvider = (props: { children?: React.ReactNode }) => {
+  const book = useSelector(bookSelector);
+  const route = useSelector(matchSelector)?.route;
+  const [polyfillLoaded, setPolyfillLoaded] = useState(false);
+  const [intl, setIntl] = useState<IntlShape | null>(null);
+
+  const bookLocale = React.useMemo(() => {
+    return route?.locale || book?.language;
```

Comment:
technically the concept of books is out of scope this high up in the app, but i'm not sure i care enough to fix it. the right encapsulation would be to defer to the route and have the content route do something with the book, but i would rather get this merged than refine this

Reply:
ok, I'll make a note to create a ticket once this is merged